python/services/dataservice/dmod/dataservice/models.py

```diff
@@ -0,0 +1,254 @@
+from enum import Enum
+from datetime import datetime
+from fastapi import UploadFile
+from pydantic.generics import GenericModel
+from pydantic import (
+    BaseModel,
+    root_validator,
+    validator,
+    StrictInt,
+    StrictFloat,
+    StrictStr,
+    Field,
+    UUID4,
+)
+from minio.datatypes import Object
+
+from dmod.core import meta_data
+from .errors import Error as ErrorEnum
+
+from typing import Dict, Optional, Union, List, Generic, TypeVar
+from typing_extensions import Self
+
+T = TypeVar("T")
+
+
+class Option(GenericModel, Generic[T]):
+    # root_validator sets this field
+    is_none: bool = True
+    value: Optional[T]
+
+    @root_validator()
+    def _enforce_none_variant(
+        cls, values: Dict[str, Union[bool, T]]
+    ) -> Dict[str, Union[bool, T]]:
+        if values["value"] is None:
+            values["is_none"] = True
+        else:
+            values["is_none"] = False
+        return values
+
+
+class Error(BaseModel):
+    type: str
+    title: str
+    status: int
+    detail: str
+
+    @classmethod
+    def from_error_enum(
+        cls, error_enum: ErrorEnum, detail: Optional[str] = None
+    ) -> Self:
+        return cls(
+            title=error_enum.name,
+            type=error_enum.type,
+            status=error_enum.status,
+            detail=detail if detail is not None else error_enum.detail,
+        )
+
+
+class StandardDatasetIndex(str, Enum):
```

Comment:
(Note that it occurred to me a little lower that perhaps I forgot that these things are temporary; if they are, then my comment here can be ignored.)

My first impression is that duplicating an implementation like this is not a good idea.  It will force us to maintain essentially the same type two places and inevitably lead to bugs periodically when we remember to update one but not the other.  

I also don't understand why it's necessary (although maybe that becomes more clear futher in the review); why not just use `dmod.core.meta_data.StandardDatasetIndex`?

My thoughts are similar for the other enum types below that are analogs of types in `dmod.core`.

I do think it would be fine if we modified those other implementations to extend `str` and `Enum`, as is done here.

Reply:
Yeah, these are temporary and can be replaced by the data structures in `dmod.core` once the pydanctic refactor has been merged. The non-pydantic classes currently in `dmod.core` dont play well with pydantic (e.g. `StandardDatasetIndex`) and, as such,  fastapi could not generated OpenApi docs without a fair amount of fuss.